pkg/build/builder/common.go

```diff
@@ -82,3 +91,141 @@ func updateBuildRevision(c client.BuildInterface, build *api.Build, sourceInfo *
 		glog.Warningf("An error occurred saving build revision: %v", err)
 	}
 }
+
+// randomBuildTag generates a random tag used for building images in such a way
+// that the built image can be referred to unambiguously even in the face of
+// concurrent builds with the same name in the same namespace.
+func randomBuildTag(namespace, name string) string {
+	repo := fmt.Sprintf("%s/%s", namespace, name)
+	randomTag := fmt.Sprintf("%08x", rand.Uint32())
+	maxRepoLen := reference.NameTotalLengthMax - len(randomTag) - 1
+	if len(repo) > maxRepoLen {
+		repo = fmt.Sprintf("%x", sha1.Sum([]byte(repo)))
+	}
+	return fmt.Sprintf("%s:%s", repo, randomTag)
+}
+
+// containerNamePrefix prefixes the name of containers launched by a build. We
+// cannot reuse the prefix "k8s" because we don't want the containers to be
+// managed by a kubelet.
+const containerNamePrefix = "openshift"
+
+// containerName creates names for Docker containers launched by a build. It is
+// meant to resemble Kubernetes' pkg/kubelet/dockertools.BuildDockerName.
```

Comment:
Hmmm... I think the guarantee we need comes in the build level. If there can be two concurrent builds of the same buildconfig, than we need the random suffix.

Anyway, if we guarantee builds run serially, the random suffix also creates no harm.

Not having the suffix, on the other hand, could lock builds if ever a temporary container failed to be removed (we throw a warning in that case).

IOW, having a random suffix allow us to keep the system healthy even under the case when, for some reason, Docker refuses to remove the temporary container.